Refactor DaPars to include mandatory run modes

faricazjj opened the pull request:
Fixes #204 

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have updated corresponding READMEs (if applicable)
- [x] My code follows the templates/style guidelines of the repository
- [x] In- and output formats comply with APAeval specifications
- [x] No parameters or file names are hardcoded
- [x] Results, logs or other output is not commited to the repository
- [x] I have tested my code

dominikburri reviewed (commented):
Thanks for the refactoring! 
I tested the execution with all 4 combinations of run modes with `samplesheet_example_files.csv` and Singularity, and each works just fine. I only have a minor comment regarding the readme.

So this part seems to be fine, but maybe it is worth fixing issue #207 with this PR as well. What do you think @faricazjj ?

faricazjj commented on execution_workflows/DaPars/README.md:
```diff
@@ -45,6 +45,8 @@ To run with singularity, comment out line 49 in Dapars/nextflow.config file `doc
 Parameters relevant to the workflow itself are:
+- `run_identification` - set to true to obtain identification challenge output. Specifying any other value will throw an error.
+- `run_differential` - set to true to obtain differential challenge output. Specifying any other value will throw an error.
 - `mode` - whether to run to obtain identification ("identification") or differential ("differential") challenge output.
```
yes you're right!

faricazjj commented:
@dominikburri Thanks so much for reviewing! :D I just addressed your comment and it's now ready for rereview!

As for the bug you reported, I created a separate PR to fix it here https://github.com/iRNA-COSI/APAeval/pull/208. I thought it would be cleaner to link a separate pr to the issue and so that it's easier to review as well   😊

yuukiiwa reviewed (approved):
This looks really good, @faricazjj!! :)
